src/modules/CurrencyConverterProxy.ts

```diff
@@ -0,0 +1,97 @@
+/*
+
+    Copyright 2020 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+import BigNumber from 'bignumber.js';
+import { Contract } from 'web3-eth-contract';
+
+import { Contracts } from './Contracts';
+import {
+  BigNumberable,
+  SendOptions,
+  TxResult,
+  address,
+} from '../lib/types';
+
+export class CurrencyConverterProxy {
+  private contracts: Contracts;
+  private proxy: Contract;
+
+  constructor(
+    contracts: Contracts,
+  ) {
+    this.contracts = contracts;
+    this.proxy = this.contracts.p1CurrencyConverterProxy;
+  }
+
+  // ============ State-Changing Functions ============
```

Comment:
Can we add getter functions for deposit/withdraw that use `this.conctracts.call` to (rather than send the transaction) simulate it and return the expected conversion amount?